src/niftyone/figures/generator.py

```diff
@@ -0,0 +1,91 @@
+"""Generator classes for different views."""
+
+import logging
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from pathlib import Path
+from typing import Any, Callable, Generic, TypeVar
+
+import nibabel as nib
+import pandas as pd
+from bids2table import BIDSEntities, BIDSTable
+from matplotlib.figure import Figure
+from PIL.Image import Image
+
+import niclips.image as noimg
+
+generator_registry: dict[str, type["ViewGenerator"]] = {}
+
+T = TypeVar("T", bound="ViewGenerator")
+
+
+def register(name: str) -> Callable[[type[T]], type[T]]:
+    """Function to register generator to registry."""
+
+    def decorator(cls: type[T]) -> type[T]:
+        generator_registry[name] = cls
+        return cls
+
+    return decorator
+
+
+def create_generators(
+    config: dict[str, Any],
+) -> list["ViewGenerator"]:
+    """Function to create generators dynamically from config."""
+    generators = []
+
+    for settings in config.values():
+        query = settings.get("query", "")
+        views = settings.get("views", [])
+
+        for view in views:
+            if view in generator_registry:
+                generator_cls = generator_registry[view]
+                generator_instance = generator_cls(query=query)
+                generators.append(generator_instance)
+            else:
+                logging.warning(f"Generator for '{view}' not found in registry")
```

Comment:
Should a missing generator be an exception not a warning?

Reply:
We could raise an exception - I guess the question is do we want the workflow to "stop" if it can't create the view?

Reply:
I've made this to just raise a custom `GeneratorError` for now within the `create_generator`call.

Reply:
I kind of prefer to just raise builtin exceptions wherever possible. Less code and one less exception class to think about handling. I would just raise `KeyError`.

Reply:
Done, happy with using the built-in exceptions. Not sure if we would have to handle any other errors at the moment. I think if there are a lot more that are related to the generator, my preference would be to go back to a custom exception.